doc/melange.rst

```diff
@@ -0,0 +1,298 @@
+:orphan:
+
+.. _melange_main:
+
+***********************************
+JavaScript Compilation With Melange
+***********************************
+
+Introduction
+============
+
+`Melange <https://github.com/melange-re/melange>`_ is a compiler from OCaml to
+JavaScript. Unlike Js_of_ocaml, Melange works by translating OCaml compiler
+internal Lambda representation to JS files. This allows to produce a single
+JavaScript file from each OCaml module. Melange can be installed with
+`opam, Esy, or Nix <https://github.com/melange-re/melange#installation>`_
+package managers.
+
+Dune can build projects using Melange, and it allows the user to generate
+JavaScript files by defining a :ref:`melange-emit` stanza. Dune libraries can be
+used with Melange by leveraging the ``(modes melange)`` field in the
+:ref:`library` stanza.
+
+To enable Melange support in a Dune project, specify the Melange language
+version in the :ref:`dune-project` file. Adding
+
+.. code:: dune
+
+    (using melange 0.1)
+
+to a :ref:`dune-project` file enables using the ``melange.emit`` stanza and the
```

Comment:
```suggestion
to a :ref:`dune-project` file enables using the ``melange.emit`` stanza to be used and the
```

Reply:
I don't get this suggestion. I am referring to the ``melange.emit`` stanza in an abstract way. I will update to:

```
to a :ref:`dune-project` file enables using ``melange.emit`` stanzas and the
``(modes melange)`` field for the :ref:`library` stanza.
```

Reply:
Sorry I meant to remove the `using` before.

Reply:
Or alternatively `enables the use of`. To my ear, `enables using` sounds strange.

Reply:
I went with `enables the use of` 👍